### ISSUE TITLE
Related indexes disappear after sqlite "removeColumn"

### ISSUE THREAD

**Issue by michael-ericson-tanium** [sequelize/sequelize#8677]
## What are you doing?

Using queryInterface.removeColumn(...) in a migration to remove an existing column from an existing table without affecting existing indexes on the target table:

```js
// Migration #1
return queryInterface.createTable('Foo', {
  id: {
    allowNull: false,
    autoIncrement: true,
    primaryKey: true,
    type: Sequelize.INTEGER,
  },
  alpha: { type: Sequelize.INTEGER },
  bravo: { type: Sequelize.INTEGER },
})
.then(function() {
  return queryInterface.sequelize.query('CREATE UNIQUE INDEX foo_alpha ON Foo (alpha)')
})

// ... Migration #2
return queryInterface.removeColumn('Foo', 'bravo');
```

## What do you expect to happen?

I expect the following objects to exist:
```sql
CREATE TABLE `Foo` (`id` INTEGER PRIMARY KEY AUTOINCREMENT, `alpha` INTEGER);
CREATE UNIQUE INDEX foo_alpha ON Foo (alpha);
```

## What is actually happening?

The database actually has the following structure - note the missing index (which was indeed present after the first migration):
```sql
CREATE TABLE `Foo` (`id` INTEGER PRIMARY KEY, `alpha` INTEGER);
```

__Dialect:__ sqlite
__Dialect version:__ npm sqlite3@3.1.13
__Database version:__ SQLite 3.16.0 2016-11-04 19:09:39 0e5ffd
__Sequelize version:__ 3.30.4
__Tested with master branch:__ YES

**Comment by sushantdhiman** [comment]
SQLite does not support ALTER TABLE DROP COLUMN statement, so we create a new table with and copy existing data to it. It seems some structure is not properly copied

**Comment by sdball** [comment]
Yes, the existing code to copy existing data does not copy over indexes: https://github.com/sdball/sequelize/blob/master/lib/dialects/sqlite/query-interface.js#L61-L74

Compared to, say, Active Record which copies over indexes to the new table: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L417

The key AR code to identify and describe the indexes: https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/sqlite3/schema_statements.rb#L8-L44
